app/src/main/java/com/osfans/trime/data/theme/mapper/ToolBarMapper.kt

```diff
@@ -0,0 +1,59 @@
+// SPDX-FileCopyrightText: 2015 - 2025 Rime community
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+package com.osfans.trime.data.theme.mapper
+
+import com.charleskorn.kaml.YamlMap
+import com.charleskorn.kaml.YamlScalar
+import com.osfans.trime.data.theme.model.ToolBar
+
+class ToolBarMapper(node: YamlMap) : Mapper<ToolBar>(node) {
+
+    override fun map(): ToolBar = ToolBar(
+        primaryButton = node.get<YamlMap>("primary_button")?.let(::mapButton),
+        buttons = getStringList("button"),
+        customButtons = node.entries.mapNotNull { (key, value) ->
+            val keyStr = key.content
+            if (keyStr !in listOf("primary_button", "button", "button_spacing", "button_font") && value is YamlMap) {
+                keyStr to mapButton(value)
```

Comment:
Here we can just decode from node:
```suggestion
                keyStr to yaml.decodeFromNode(value)
```
and custom mappings below could be removed.